src/Microsoft.CodeQuality.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.Fixer.cs

```diff
@@ -0,0 +1,96 @@
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeQuality.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1827: Do not use Count() when Any() can be used.
+    /// </summary>
+    public abstract class DoNotUseCountWhenAnyCanBeUsedFixer : CodeFixProvider
+    {
+        /// <summary>
+        /// A list of diagnostic IDs that this provider can provider fixes for.
+        /// </summary>
+        /// <value>The fixable diagnostic ids.</value>
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(DoNotUseCountWhenAnyCanBeUsedAnalyzer.RuleId);
+
+        /// <summary>
+        /// Gets an optional <see cref="FixAllProvider" /> that can fix all/multiple occurrences of diagnostics fixed by this code fix provider.
+        /// Return null if the provider doesn't support fix all/multiple occurrences.
+        /// Otherwise, you can return any of the well known fix all providers from <see cref="WellKnownFixAllProviders" /> or implement your own fix all provider.
+        /// </summary>
+        /// <returns>FixAllProvider.</returns>
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <summary>
+        /// Computes one or more fixes for the specified <see cref="CodeFixContext" />.
+        /// </summary>
+        /// <param name="context">A <see cref="CodeFixContext" /> containing context information about the diagnostics to fix.
+        /// The context must only contain diagnostics with a <see cref="Diagnostic.Id" /> included in the <see cref="CodeFixProvider.FixableDiagnosticIds" /> 
+        /// for the current provider.</param>
+        /// <returns>A <see cref="Task" /> that represents the asynchronous operation.</returns>
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(context.Span);
+
+            if (this.TryGetFixer(node, out var expression, out var arguments, out var negate))
+            {
+                var title = MicrosoftCodeQualityAnalyzersResources.DoNotUseCountWhenAnyCanBeUsedTitle;
+                context.RegisterCodeFix(
+                    CodeAction.Create(
```

Comment:
We use a privately defined code action type, such as https://github.com/dotnet/roslyn-analyzers/blob/32d8f1e397439035f0ecb5f61a9e672225f0ecdb/src/Microsoft.NetCore.Analyzers/Core/Runtime/AvoidUnsealedAttributes.Fixer.cs#L51, primarily for telemetry purposes. See https://github.com/dotnet/roslyn/issues/4919 for details.